src/neo/Network/P2P/TaskManager.cs

```diff
@@ -109,7 +109,7 @@ private void OnNewTasks(InvPayload payload)
         private void OnPersistCompleted(Block block)
         {
             foreach (var (actor, session) in sessions)
-                if (session.ReceivedBlock.Remove(block.Index))
+                if (session.ReceivedBlock.Remove(block.Index, out Block receivedBlock) && block.Hash == receivedBlock.Hash)
```

Comment:
What's the different? The invalid one has been removed by `OnInvalidBlock()`.

Reply:
I think there might also be cases where the relayed result in `UnableToVerify` but later found out is invalid

Reply:
But we still need to request task for that node, right?

Reply:
Maybe we should not trust a node which has sent an invalid block?